From f9cac810f554b37aa87d465c7f035eff23657985 Mon Sep 17 00:00:00 2001
From: Codex <codex@local.invalid>
Date: Sun, 16 Aug 2026 22:48:58 +0800
Subject: [PATCH] Read ROM metadata from system build properties only

---
 FoxExtras/sbin/foxstart.sh | 61 ++++++--------------------------------
 1 file changed, 9 insertions(+), 52 deletions(-)

diff --git a/FoxExtras/sbin/foxstart.sh b/FoxExtras/sbin/foxstart.sh
index ec1580e..3efb4a4 100755
--- a/FoxExtras/sbin/foxstart.sh
+++ b/FoxExtras/sbin/foxstart.sh
@@ -175,9 +175,7 @@ is_SAR() {
 get_ROM() {
 local S="/tmp_system_rom"
 local PROP="$S/build.prop"
-local V="/vendor"
-local V_PROP="$V/build.prop"
-local found_vendor_prop="0"
+
 local slot=$(getprop "ro.boot.slot_suffix")
 
    # don't use slots when using a dm-* block device name
@@ -220,34 +218,11 @@ local slot=$(getprop "ro.boot.slot_suffix")
    umount $S > /dev/null 2>&1
    rmdir $S > /dev/null 2>&1
   
-   # now look for vendor prop
-   local mv1="0"
-   [ ! -d "$V" ] && mkdir -p $V > /dev/null 2>&1
-   if [ -d "$V" ]; then
-      ! is_mounted $V && {
-         $MOUNT_CMD $V > /dev/null 2>&1
-         is_mounted $V && mv1="1"
-      }
-      is_mounted $V && {
-         [ -f $V_PROP ] && {
-           found_vendor_prop=1
-           # make a copy of the vendor build.prop
-           cp $V_PROP "$OUR_TMP/vendor_build_prop" > /dev/null 2>&1
-           V_PROP="$OUR_TMP/vendor_build_prop"
-           [ "$mv1" = "1" ] && umount $V > /dev/null 2>&1
-         }
-      }
-   fi
-   
-   # use the vendor prop if found
-   [ "$found_vendor_prop" != "1" ] && V_PROP=""
 
    # query the build.prop
    local tmp2=""
    tmp2=$(file_getprop "$PROP" "ro.build.display.id")
-   [ -n "$V_PROP" -a  -z "$tmp2" ] && tmp2=$(file_getprop "$V_PROP" "ro.vendor.build.id")
-   [ -z "$tmp2" ] && tmp2=$(file_getprop "$PROP" "ro.build.id")
-   [ -z "$tmp2" ] && tmp2=$(file_getprop "$PROP" "ro.system.build.id")
+
    
    # ROM not found?
    if [ -z "$tmp2" ]; then
@@ -259,9 +234,7 @@ local slot=$(getprop "ro.boot.slot_suffix")
    # we have a ROM - get SDK, etc
    local tmp3=""
    if [ -n "$tmp2" ]; then
-      [ -n "$V_PROP" ] && tmp3=$(file_getprop "$V_PROP" "ro.vendor.build.version.sdk") 
-      [ -z "$tmp3" ] && ttmp3=$(file_getprop "$PROP" "ro.build.version.sdk")
-      [ -z "$tmp3" ] && tmp3=$(file_getprop "$PROP" "ro.system.build.version.sdk")
+      tmp3=$(file_getprop "$PROP" "ro.build.version.sdk")
       [ -n "$tmp3" ] && {
          ANDROID_SDK="$tmp3"
          $SETPROP orangefox.rom.sdk "$tmp3" > /dev/null 2>&1
@@ -270,29 +243,19 @@ local slot=$(getprop "ro.boot.slot_suffix")
       }
 
       # get incremental version
-      [ -n "$V_PROP" ] && tmp3=$(file_getprop "$V_PROP" "ro.vendor.build.version.incremental")
-      [ -z "$tmp3" ] && tmp3=$(file_getprop "$PROP" "ro.build.version.incremental")
-      [ -z "$tmp3" ] && tmp3=$(file_getprop "$PROP" "ro.system.build.version.incremental")
+      tmp3=$(file_getprop "$PROP" "ro.build.version.incremental")
       [ -n "$tmp3" ] && {
         echo "DEBUG: OrangeFox: INCREMENTAL_VERSION=$tmp3" >> $LOG
         echo "INCREMENTAL_VERSION=$tmp3" >> $CFG
-        [ -x "$SETPROP" ] && {
-              $SETPROP "ro.build.version.incremental" "$tmp3" > /dev/null 2>&1
-              $SETPROP "orangefox.system.incremental" "$tmp3" > /dev/null 2>&1
-        }
+        [ -x "$SETPROP" ] && $SETPROP "orangefox.system.incremental" "$tmp3" > /dev/null 2>&1
       }
 
       # get "release" version
-      [ -n "$V_PROP" ] && tmp3=$(file_getprop "$V_PROP" "ro.vendor.build.version.release")
-      [ -z "$tmp3" ] && tmp3=$(file_getprop "$PROP" "ro.build.version.release")
-      [ -z "$tmp3" ] && tmp3=$(file_getprop "$PROP" "ro.system.build.version.release")
+      tmp3=$(file_getprop "$PROP" "ro.build.version.release")
       [ -n "$tmp3" ] && {
         echo "DEBUG: OrangeFox: RELEASE_VERSION=$tmp3" >> $LOG
         echo "RELEASE_VERSION=$tmp3" >> $CFG
-        [ -x "$SETPROP" ] && {
-              $SETPROP "ro.build.version.release" "$tmp3" > /dev/null 2>&1
-              $SETPROP "orangefox.system.release" "$tmp3" > /dev/null 2>&1
-        }
+        [ -x "$SETPROP" ] && $SETPROP "orangefox.system.release" "$tmp3" > /dev/null 2>&1
       }
       
       # and other stuff
@@ -306,17 +269,11 @@ local slot=$(getprop "ro.boot.slot_suffix")
    # check for ROM fingerprints
    local FP=""
    if [ -n "$tmp2" ]; then
-      [ -n "$V_PROP" ] && FP=$(file_getprop "$V_PROP" "ro.vendor.build.fingerprint")
-      [ -z "$FP" ] && FP=$(file_getprop "$PROP" "ro.build.version.base_os")
-      [ -z "$FP" ] && FP=$(file_getprop "$PROP" "ro.build.fingerprint")
-      [ -z "$FP" ] && FP=$(file_getprop "$PROP" "ro.system.build.fingerprint")
+      FP=$(file_getprop "$PROP" "ro.build.fingerprint")
       [ -n "$FP" ] && {
            echo "ROM_FINGERPRINT=$FP" >> $CFG
            echo "DEBUG: OrangeFox: ROM_FINGERPRINT=$FP" >> $LOG
-           [ -x "$SETPROP" ] && {
-              $SETPROP "ro.build.fingerprint" "$FP" > /dev/null 2>&1
-              $SETPROP "orangefox.system.fingerprint" "$FP" > /dev/null 2>&1
-            }
+           [ -x "$SETPROP" ] && $SETPROP "orangefox.system.fingerprint" "$FP" > /dev/null 2>&1
       }
    fi # check for ROM fingerprints
    
-- 
2.47.0.windows.2

